domain-model-assistant/Assets/Components/Scripts/Diagram.cs

```diff
@@ -11,419 +11,549 @@
 using UnityEngine.Networking;
 using UnityEngine.UI;
 
+
 public class Diagram : MonoBehaviour
 {
 
-  public TextAsset jsonFile;
-  public float zoomSpeed = 1;
-  public CanvasScaler CanvasScaler;
-  public float targetOrtho;
-  public float smoothSpeed = 2.0f;
-  public float minOrtho = 0.0f;
-  public float maxOrtho = 20.0f;
-  private Vector3 dragStartPos;
-  // private bool dragging = false;
-  public GameObject compartmentedRectangle;
-  public List<GameObject> compartmentedRectangles;
+    public TextAsset jsonFile;
+    public float zoomSpeed = 1;
+    public CanvasScaler CanvasScaler;
+    public float targetOrtho;
+    public float smoothSpeed = 2.0f;
+    public float minOrtho = 0.0f;
+    public float maxOrtho = 20.0f;
+    private Vector3 dragStartPos;
+    // private bool dragging = false;
+    public GameObject compartmentedRectangle;
+    public List<GameObject> compartmentedRectangles;
+
+    public const bool UseWebcore = true; // Change to false to use the wrapper page JSON instead of WebCore
+
+    public const string WebcoreEndpoint = "http://localhost:8080/";
 
-  public const bool UseWebcore = true; // Change to false to use the wrapper page JSON instead of WebCore
+    public const string cdmName = "MULTIPLE_CLASSES"; // TODO Remove this once API is updated
 
-  public const string WebcoreEndpoint = "http://localhost:8080/";
+    public const string GetCdmEndpoint = WebcoreEndpoint + "classdiagram/" + cdmName;
 
-  public const string cdmName = "MULTIPLE_CLASSES"; // TODO Remove this once API is updated
+    public const string AddClassEndpoint = GetCdmEndpoint + "/class";
 
-  public const string GetCdmEndpoint = WebcoreEndpoint + "classdiagram/" + cdmName;
+    public const string DeleteClassEndpoint = AddClassEndpoint; // + "/:class_id"
 
-  public const string AddClassEndpoint = GetCdmEndpoint + "/class";
+    public const string UpdateClassEndpoint = GetCdmEndpoint; //+ {classId}/position
+    private ClassDiagramDTO classDTO;
 
-  public const string DeleteClassEndpoint = AddClassEndpoint; // + "/:class_id"
+    GraphicRaycaster raycaster;
 
-  private ClassDiagramDTO classDTO;
+    Dictionary<string, GameObject> _namesToRects = new Dictionary<string, GameObject>();
 
-  GraphicRaycaster raycaster;
+    Dictionary<string, List<Attribute>> idsToClassesAndAttributes = new Dictionary<string, List<Attribute>>();
```

Comment:
Variable name does not go well with type.

Reply:
here was updated